src/terrain/routes/schemas/resource_usage_api.clj

```diff
@@ -0,0 +1,117 @@
+   :user_id         (describe UUID "The UUID assigned to the user the total applies to")
+   :username        (describe String "The username of the user the total applies to")
+   :total           (describe Integer "The total number of CPU hours that the user has reserved")
+   :effective_start (describe String "The start date for the time range during which the total is applicable")
+   :effective_end   (describe String "The end date for the time range during which the total is applicable ")
+   :last_modified   (describe String "The date the record in the database was last modified")})
+
+(defschema Event
+  {:id                      EventID
+   :record_date             (describe String "The date the event was added to the system")
+   :effective_date          (describe String "The date the event is effective")
+   :event_type              (describe String "The type of event")
+   :value                   (describe Integer "The value associated with the event. How it's used is determined by the event type")
+   :created_by              (describe UUID "The UUID of the user that created the event")
+   :claimed                 (describe Boolean "Whether the event has been claimed by a worker")
+   :claimed_by              (describe (maybe UUID) "The UUID of the worker that claimed the event")
+   :claim_expires_on        (describe (maybe String) "The date the worker's claim expires")
+   :claimed_on              (describe (maybe String) "The date the worker claimed the event")
+   :processed               (describe Boolean "Whether the event has been worked on by a worker")
+   :processing              (describe Boolean "Whether a worker is currently processing the event")
+   :processed_on            (describe (maybe String) "The date the worker finished processing the event")
+   :max_processing_attempts (describe Integer "The number of times any worker can try to process the event")
+   :attempts                (describe Integer "The number of times any worker has tried to process the event")
+   :last_modified           (describe String "The last time the record was modified in the system")})
+
+(defschema UpdateEvent
+  (into {}
+        (for [keyval (select-keys Event [:record_date
+                                         :effective_date
+                                         :event_type
+                                         :value
+                                         :created_by
+                                         :claimed
+                                         :claimed_by
+                                         :claim_expires_on
+                                         :claimed_on
+                                         :processed
+                                         :processing
+                                         :max_processing_attempts
+                                         :attempts])]
+          [(optional-key (key keyval)) (val keyval)])))
```

Comment:
I think this also could be handled by `schema-tools.core/optional-keys`.

Reply:
Oops, missed that. Thanks!